pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,140 @@
+package agent
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+
+	// Provider contains settings for different providers for which an agent based cluster installation
+	// can be performed.
+	Provider Provider `json:"provider"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	// TODO: should this be optional and defaulted?
+	Networking Networking `json:"networking"`
```

Comment:
We propose to keep this one on ClusterDeployment as it is per cluster, not a good fit for InstallEnvironment and does not go into an installation ISO.